engine/rk56.go

```diff
@@ -39,73 +50,105 @@ func (rk *RK56) Step() {
 	h := float32(Dt_si * GammaLL) // internal time step = Dt * gammaLL
 
 	// stage 1
-	torqueFn(k1)
+	torqueFn(rk.k1)
```

Comment:
This comment is dependent on the outcome of the comment further down, about the final torque evaluation.
The Runge-Kutta-Fehlberg solver does not inherently have the First-Same-As-Last property. The reason that `k2` can be recycled as `rk.k1` here, is because the torque acting on the final magnetization state is calculated anyway to allow calculating `SetMaxTorque()`. Hence, dependent on how we choose to deal with the inconsistency below:
- If we choose to use the final torque for `SetMaxTorque()`, then we can indeed cache that final torque evaluation for the next iteration, and this can then similarly be done for all solvers even if they do not have the FSAL property.
- If we choose to use the torque of the last RK stage for `SetMaxTorque()`, then we can not cache it for the next step and we fall back to only being able to re-use `rk.k1` for FSAL solvers (i.e. can not use `data.Copy()`).